src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -14,6 +15,9 @@ namespace Microsoft.AspNetCore.Razor.Parser.TagHelpers.Internal
 {
     public class TagHelperParseTreeRewriter : ISyntaxTreeRewriter
     {
+        // Newlines are invalid markup for HTML attribute values.
+        private static readonly string InvalidAttributeValueMarker = Environment.NewLine;
```

Comment:
1. Newlines are not disallowed in an HTML attribute value. They're usually escaped e.g. `&xA;` but aren't required to be.
2. There is no requirement that HTML matches the system's line endings e.g. `&xD;&xA;` should be fine on a Unix system.